clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
@ijuma @junrao Thanks much. This makes sense. I have updated the patch to include `UnknownRetriableException`, bumped up the version of ProduceRequest and FetchRequest, and converted KafkaStorageException to the error code of NotLeaderForPartitionException for existing versions of ProduceResponse and FetchResponse.